sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationSuite.scala

```diff
@@ -501,4 +501,28 @@ class HiveScriptTransformationSuite extends BaseScriptTransformationSuite with T
       """.stripMargin)
     checkAnswer(query4, identity, Row(null) :: Nil)
   }
+
+  test("SPARK-32684: Script transform hive serde mode null format is same with hive as '\\N'") {
+    val query1 = sql(
+      """
+        |SELECT TRANSFORM(null, null, null)
+        |USING 'cat'
+        |FROM (SELECT 1 AS a, 2 AS b, 3 AS c) t
+      """.stripMargin)
+    checkAnswer(query1, identity, Row(null, "\\N\t\\N") :: Nil)
+
+    val query2 = sql(
+      """
+        |SELECT TRANSFORM(null, null, null)
+        |  ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe'
+        |USING 'cat'
+        |  ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe'
+        |  WITH SERDEPROPERTIES (
+        |   'serialization.last.column.takes.rest' = 'true'
```

Comment:
Why do we need this conf in this test?

Reply:
> Why do we need this conf in this test?

Use schemaless mode and use this value to make `value` column take all rest data and show the `\\N`, or it will just show `null`

Reply:
> Why do we need this conf in this test?

Updated, more clear now?